android/src/main/java/bbct/android/common/activity/MainActivity.java

```diff
@@ -62,25 +65,24 @@ public void onCreate(Bundle savedInstanceState) {
         }
 
         this.setContentView(R.layout.main);
+        fragmentManager = getSupportFragmentManager();
 
         if (savedInstanceState == null) {
             Uri uri = BaseballCardContract.getUri(this.getPackageName());
             Cursor cursor = this.getContentResolver().query(uri,
                     BaseballCardContract.PROJECTION, null, null, null);
+            fragmentManager.beginTransaction().add(R.id.fragment_holder, new BaseballCardList(), FragmentTags.CARD_LIST)
+                    .addToBackStack(FragmentTags.CARD_LIST)
+                    .commit();
 
-            FragmentTransaction ft = this.getSupportFragmentManager().beginTransaction();
             if (cursor == null || cursor.getCount() == 0) {
-                ft.add(R.id.fragment_holder, new BaseballCardDetails(), FragmentTags.EDIT_CARD);
-            } else {
-                ft.add(R.id.fragment_holder, new BaseballCardList(), FragmentTags.CARD_LIST);
-                cursor.close();
+                fragmentManager.beginTransaction().replace(R.id.fragment_holder, new BaseballCardDetails(), FragmentTags.EDIT_CARD)
+                        .addToBackStack(FragmentTags.EDIT_CARD)
+                        .commit();
             }
-            ft.commit();
```

Comment:
Check if something like this will work (pseudocode):

    begin fragment transaction
    add list fragment
    if database is empty
        replace list fragment with details fragment
    commit transaction

I'm not sure if adding two different fragments with the same transaction will work. If it does, the code will read better because you can remove the multiple calls to `beginTransaction()`.

Reply:
@codeguru42 implementing this breaks the app. 
It results into:
MainActivity sets BaseballCardDetails as initial fragment, and BaseballCardList is lost.

Reply:
Okay, we'll leave it as it is. Please break the `add()` and `replace()` calls on a new line.

Reply:
Okay I've pushed an update for this.